### PR TITLE
Create redis-cluster.js

### PR DESCRIPTION
**Pull request by zahmedFDL**
AWS Elasticache Clustered Redis

### CODE REVIEW

File: lib/cache/stores/redis-cluster.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const _         = require('lodash');
```
Comment (patrixr) [inline lib/cache/stores/redis-cluster.js:1]: @derrickmehaffy You've worked on Redis Cluster support previously haven't you ? Do you think this new store is warranted ? Or can the existing redis store be used ?

Comment (derrickmehaffy) [reply lib/cache/stores/redis-cluster.js:1]: I'll have to do some testing, I did briefly mess with clusters and the main issue was the key partitioning, sentinels work fine.

**Comment by derrickmehaffy** [comment]
@zahmedFDL have you tested this on standard redis clusters or only AWS elasticache?

**Comment by zahmedFDL** [comment]
@derrickmehaffy   on AWS elasticache only.

I have updated the code redis-cluster.js  to allow the user to add more options into config/middleware.js 

`cache: {
      enabled: true,
      type: 'redis-cluster',
      maxAge: 3600000,
      models: [
        {
          model: 'Articles',
          maxAge: 1000000,
        },
        {
          model: 'Writers',
          maxAge: 1000000,
        },
        {
          model: 'Categories',
          maxAge: 1000000,
        }
      ],
      redisConfig: {
	   options: {
		slotsRefreshTimeout: 5000,
		dnsLookup: (address, callback) => callback(null, address),
		slotsRefreshTimeout: 1000,   
	   },
	   startupNodes: [{
    		port: 6379,
    		host: 'hidden.cache.amazonaws.com',
  	   }]
       },
    }`

**Comment by derrickmehaffy** [comment]
> @derrickmehaffy on AWS elasticache only.
> 
> I have updated the code redis-cluster.js to allow the user to set more options into config/middleware.js
> 
> `cache: { enabled: true, type: 'redis-cluster', maxAge: 3600000, models: [ { model: 'Articles', maxAge: 1000000, }, { model: 'Writers', maxAge: 1000000, }, { model: 'Categories', maxAge: 1000000, } ], redisConfig: { options: { slotsRefreshTimeout: 5000, dnsLookup: (address, callback) => callback(null, address), slotsRefreshTimeout: 1000, }, startupNodes: [{ port: 6379, host: 'hidden.cache.amazonaws.com', }] }, }`

We will want to test on an actual Redis Cluster before merging. Elasticache is no longer entirely Redis based, as Redis changed their license since Amazon wasn't contributing their changes back to them. Amazon has modified Redis and it's clustering in some very fundamental ways that differ from standard Redis.

Specifically this FAQ entry from: https://redislabs.com/legal/licenses/


> Why did Redis Labs adopt RSAL?
> Modern open source infrastructure software has created more value over the past decade than we could have ever imagined. Databases, orchestrators, distributed systems and other software technologies now power nearly every business on the planet — all thanks to the shared, collaborative philosophy of the open source community.
> 
> However, some cloud providers have repeatedly taken advantage of successful open source projects, without significant contributions to their communities. They repackage software that was not developed by them into competitive, proprietary service offerings and use their business leverage to reap substantial revenues from these open source projects.
> 
> Redis Labs has been leading and financing the development of open source Redis for years, and at the same time believe we deserve the fruits of these efforts. While the Redis core is and will always remain available under the open source BSD license, in order to keep our business and the Redis project sustainable, we’ve decided to license certain modules built by Redis Labs (e.g. RediSearch, RedisGraph, RedisJSON, RedisBloom, RedisML) with the Redis Source Available License.

**Comment by zahmedFDL** [comment]
@derrickmehaffy  Maybe we can rename it to **aws-elasticache** instead or **redis-cluster** for more clarity. 

The goal is to at least provide a solution for the ppl that are using was elasticache.

What do you think?

Regards,

Zaki

**Comment by derrickmehaffy** [comment]
We should be able to handle both so long as the lib supports it, but we need to test it.

**Pull request closed without merging**